src/compiler/compile/render_dom/Renderer.ts

```diff
@@ -248,7 +248,7 @@ export default class Renderer {
 						.reduce((lhs, rhs) => x`${lhs} | ${rhs}`);
 				}
 
-				return x`${dirty} & /*${names.join(', ')}*/ ${bitmask[0].n}` as BinaryExpression;
+				return x`${dirty}[0] & /*${names.join(', ')}*/ ${bitmask[0].n}` as BinaryExpression;
```

Comment:
If we're not doing the destructuring, do we even need a separate case for this now? (And do we need the concept of `context_overflow`?)

Reply:
oh a good point, i didnt know what else `context_overflow` is for, let me do a global search and understand them... 🤔

Reply:
If I remove the check for `context_overflow` here and always act like there was an overflow, the tests still pass. It looks like the other place that `context_overflow` is used is in conditionally constructing an array of `-1`s to pass to the `@init` helper. If nothing is passed, that argument defaults to `[-1]`. I think that makes sense to continue to be a default (since a lot of the time there will be at most 31 things to track), but I think it would make sense to just do the <= 31 check there where we're (conditionally) filling the array, and get rid of `context_overflow`.

Reply:
yup let me update them